### PR TITLE
XW-3997 | disable indexing articles for wikis with wgDisableShowInRecirculation set to true

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-3997

### CODE REVIEW

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 1]
```diff
@@ -61,18 +63,17 @@ public static function onOasisSkinAssetGroups( &$jsAssets ) {
...
 
-		$showableNameSpaces = array_merge( $wg->ContentNamespaces, [ NS_FILE ] );
+		$showableNameSpaces = array_merge( $wg->ContentNamespaces, [ NS_FILE, NS_BLOG_ARTICLE ] );
 
-		if ( $wg->Title->exists() &&
-		     in_array( $wg->Title->getNamespace(), $showableNameSpaces ) &&
-		     $wg->request->getVal( 'action', 'view' ) === 'view' &&
-		     $wg->request->getVal( 'diff' ) === null &&
-		     !WikiaPageType::isCorporatePage()
+		if ( ( $wg->Title->exists() && in_array( $wg->Title->getNamespace(), $showableNameSpaces ) ) &&
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecirculationHooks.class.php#L74](https://github.com/Wikia/app/blob/5fa2d109b28054acc5c8c575c522deabcf4a9482/extensions/wikia/Recirculation/RecirculationHooks.class.php#L74): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 2]
```diff
@@ -163,12 +167,10 @@ private static function isPrivateOrProduction() {
...
 
-	private static function isCorrectNameSpace() {
-		$context = RequestContext::getMain();
-		$title = $context->getTitle();
+	private static function isNoIndexNamespace() {
+		$title = RequestContext::getMain()->getTitle();
 
-		return $title->inNamespace( NS_FILE ) ||
-			( $title->inNamespace( NS_BLOG_ARTICLE ) && empty( $metaDataFromService ) );
+		return $title->inNamespace( NS_FILE ) || $title->inNamespace( NS_BLOG_ARTICLE );
```
Comment (rybmat) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: how about `$title->inNamespaces(NS_FILE, NS_BLOG_ARTICLE)

Comment (rybmat) [reply extensions/wikia/Recirculation/RecirculationHooks.class.php]: and in that case the above variable assignment is redundant

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 3]
```diff
@@ -150,11 +150,15 @@ private static function checkIfIsProduction() {
 			$wgWikiaEnvironment !== WIKIA_ENV_STAGING;
 	}
 
-	private static function shouldIndex() {
-		return self::isPrivateOrProduction() || self::isCorrectNameSpace();
+	private static function shoudlNoIndex( $metaDataFromService ) {
```
Comment (rybmat) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php:155]: how about naming it `shouldIndex` and negating the returned value? IMO names without negation are easier to read

**Review by rybmat — CHANGES_REQUESTED** [review]

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 4]
```diff
@@ -70,11 +70,13 @@ static public function isCorrectPageType() {
 		$wg = F::app()->wg;
 
 		$showableNameSpaces = array_merge( $wg->ContentNamespaces, [ NS_FILE, NS_BLOG_ARTICLE ] );
+		$isInShowableNamespace =
+			$wg->Title->exists() && in_array( $wg->Title->getNamespace(), $showableNameSpaces );
```
Comment (rybmat) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: `$wg->Title->inNamespaces($showableNamespaces)`?

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 5]
```diff
@@ -70,11 +70,13 @@ static public function isCorrectPageType() {
...
+		$isViewAction =
```
Comment (rybmat) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: https://github.com/Wikia/app/blob/e5017b78863434e52eb0b9db12103ce20c886dd6/includes/wikia/WikiaPageType.class.php#L217

File: extensions/wikia/Recirculation/RecirculationHooks.class.php (hunk 4 above)
Comment (rybmat) [reply extensions/wikia/Recirculation/RecirculationHooks.class.php]: and why do you get title via $wg->Title and in other place from RequestContext? please unify

**Review by rybmat — CHANGES_REQUESTED** [review]

**Pull request merged**